### ISSUE TITLE
MD Generation

### ISSUE THREAD

**Issue by kentcb** [geaz/sharpDox#24]
You know what would be freaking amazing? If sharpDox could generate GitHub-compatible markdown files. That way I could generate markdown from my API docs and commit it them to my repo, which would make them automatically browsable on GitHub.

Just a thought.

**Comment by nikeee** [comment]
It would be even cooler if it could compile automatically to a full-static gh-pages site (using Jekyll?).

**Comment by Geaz** [comment]
Why not use the current HTML output for gh-pages? I am doing the same at http://sharpdox.de/en/doc/.

**Comment by GeertvanHorrik** [comment]
I am working towards a combination of SharpDox and DocNet:

1. Write custom stuff using DocNet
2. Generate the API stuff to markdown using SharpDox
3. DocNet will automatically include the reference documentation

For now the only downside I found in the markdown generation is that single line ode blocks are also decorated with ``` instead of `

Would you accept a PR that would fix this for single line code blocks?

For example, I want to be able to tell something is `null`.

**Comment by geaz** [comment]
Sure, go for it :)

One other question: What feature of DocNet are you missing in sharpDox? Just want to know how I could improve the feature set in the upcoming version 👍

**Comment by GeertvanHorrik** [comment]
Good point, not sure. I thought sharpdox was just for API docs and I already heavily invested in DocNet (although I must say most of the work was converting confluence => markdown). DocNet has a few nice features such as the alert boxes:

http://fransbouma.github.io/DocNet/markdownextensions.htm#alert-boxes

I've written a custom exporter so SharpDox can be used together with DoxNet:

https://github.com/CatenaLogic/SharpDox.Plugins.DocNet

It's still WIP but starts to take some shape:

http://docs.catelproject.com/vnext/reference/Catel.Serialization.Json/Catel/Runtime/Serialization/Json/CatelJsonConverter.htm
